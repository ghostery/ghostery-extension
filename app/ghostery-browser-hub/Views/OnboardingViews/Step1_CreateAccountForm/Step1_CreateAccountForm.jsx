/**
 * Browser Create Account Form
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2021 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';
import ToggleCheckbox from '../../../../shared-components/ToggleCheckbox';

/**
 * A Functional React component for rendering the Browser Create Account View
 * @return {JSX} JSX for rendering the Browser Create Account View of the Hub app
 * @memberof GhosteryBrowserHubViews
 */
export const Step1_CreateAccountForm = (props) => {
	const {
		email,
		emailError,
		confirmEmail,
		confirmEmailError,
		firstName,
		lastName,
		password,
		passwordInvalidError,
		passwordLengthError,
		confirmPassword,
		confirmPasswordError,
		legalConsentChecked,
		legalConsentNotCheckedError,
		handleLegalConsentCheckboxChange,
		handleInputChange,
		handleSubmit,
	} = props;

	const emailInputClassNames = ClassNames('Step1_CreateAccountForm__inputBox', {
		error: emailError,
	});
	const confirmInputClassNames = ClassNames('Step1_CreateAccountForm__inputBox', {
		error: confirmEmailError,
	});
	const passwordInputClassNames = ClassNames('Step1_CreateAccountForm__inputBox', {
		error: passwordInvalidError || passwordLengthError,
	});
	const legalConsentClassNames = ClassNames('Step1_CreateAccountForm__legalConsentCheckedLabel', {
		error: legalConsentNotCheckedError
	});

	return (
		<form onSubmit={handleSubmit}>
			<div className="Step1_CreateAccountForm--addPaddingTop row align-center-middle">
				<div className="columns small-10 medium-6">
					<label htmlFor="create-account-email" className="Step1_CreateAccountForm__inputLabel">
						{t('email_colon')}
					</label>
					<input
						id="create-account-email"
						className={emailInputClassNames}
						name="email"
						type="text"
						value={email}
						onChange={handleInputChange}
						pattern=".{1,}"
						autoComplete="off"
						placeholder="example@mail.com"
					/>
					{emailError && (
						<div className="Step1_CreateAccountForm__inputErrorContainer">
							<div className="Step1_CreateAccountForm__inputError">
								{t('please_enter_a_valid_email')}
							</div>
						</div>
					)}
				</div>
				<div className="columns small-10 medium-6">
					<label htmlFor="create-account-confirmEmail" className="Step1_CreateAccountForm__inputLabel">
						{t('confirm_email_colon')}
					</label>
					<input
						id="create-account-confirmEmail"
						className={confirmInputClassNames}
						name="confirmEmail"
						type="text"
						value={confirmEmail}
						onChange={handleInputChange}
						pattern=".{1,}"
						autoComplete="off"
					/>
					{confirmEmailError && (
						<div className="Step1_CreateAccountForm__inputErrorContainer">
							<div className="Step1_CreateAccountForm__inputError">
								{t('your_email_do_not_match')}
							</div>
						</div>
					)}
				</div>
			</div>
			<div className="row align-center-middle">
				<div className="columns small-10 medium-6">
					<label htmlFor="create-account-firstName" className="Step1_CreateAccountForm__inputLabel">
						{t('hub_create_account_label_first_name')}
					</label>
					<input
						id="create-account-firstName"
						className="Step1_CreateAccountForm__inputBox"
						name="firstName"
						type="text"
						value={firstName}
						onChange={handleInputChange}
						pattern=".{1,}"
						autoComplete="off"
					/>
				</div>
				<div className="columns small-10 medium-6">
					<label htmlFor="create-account-lastName" className="Step1_CreateAccountForm__inputLabel">
						{t('hub_create_account_label_email_last_name')}
					</label>
					<input
						id="create-account-lastName"
						className="Step1_CreateAccountForm__inputBox"
						name="lastName"
						type="text"
						value={lastName}
						onChange={handleInputChange}
						pattern=".{1,}"
						autoComplete="off"
					/>
				</div>
			</div>
			<div className="row align-center-middle">
				<div className="columns small-10 medium-6">
					<label htmlFor="create-account-password" className="Step1_CreateAccountForm__inputLabel">
						{t('password_colon')}
					</label>
					<input
						id="create-account-password"
						className={passwordInputClassNames}
						name="password"
						type="password"
						value={password}
						onChange={handleInputChange}
						pattern=".{1,}"
						autoComplete="off"
					/>
					{passwordInvalidError && (
						<div className="Step1_CreateAccountForm__inputErrorContainer">
							<div className="Step1_CreateAccountForm__inputError">
								{t('hub_create_account_label_password_invalid')}
							</div>
						</div>
					)}
					{passwordLengthError && (
						<div className="Step1_CreateAccountForm__inputErrorContainer">
							<div className="Step1_CreateAccountForm__inputError">
								{t('hub_create_account_label_password_invalid_length')}
							</div>
						</div>
					)}
				</div>
				<div className="columns small-10 medium-6">
					<label htmlFor="create-account-password" className="Step1_CreateAccountForm__inputLabel">
						{t('confirm_password_colon')}
					</label>
					<input
						id="create-account-password"
						className={passwordInputClassNames}
						name="confirmPassword"
						type="password"
						value={confirmPassword}
						onChange={handleInputChange}
						pattern=".{1,}"
						autoComplete="off"
					/>
					{confirmPasswordError && (
						<div className="Step1_CreateAccountForm__inputErrorContainer">
							<div className="Step1_CreateAccountForm__inputError">
								{t('hub_create_account_confirm_password_do_not_match')}
							</div>
						</div>
					)}
				</div>
			</div>
			<div className="row">
				<div className="columns small-12 medium-10">
					<div className="Step1_CreateAccountForm__checkboxContainer BrowserCreateAccountForm--marginBottom flex-container">
						<ToggleCheckbox
							name="globals"
							checked={legalConsentChecked}
							className="ToggleCheckbox--flush-left"
							onChange={handleLegalConsentCheckboxChange}
						/>
						<span
							className={legalConsentClassNames}
							onClick={handleLegalConsentCheckboxChange}
							dangerouslySetInnerHTML={{ __html: t('create_account_form_legal_consent_checkbox_label') }}
						/>
					</div>
				</div>
			</div>
			<div className="Step1_CreateAccountForm__ctaButtonContainer">
				<button type="submit" className="Step1_CreateAccountForm__ctaButton">{t('create_account')}</button>
			</div>
		</form>
	);
};

// PropTypes ensure we pass required props of the correct type
Step1_CreateAccountForm.propTypes = {
	email: PropTypes.string.isRequired,
	emailError: PropTypes.bool.isRequired,
	confirmEmail: PropTypes.string.isRequired,
	confirmEmailError: PropTypes.bool.isRequired,
	firstName: PropTypes.string.isRequired,
	lastName: PropTypes.string.isRequired,
	password: PropTypes.string.isRequired,
	confirmPassword: PropTypes.string.isRequired,
	passwordInvalidError: PropTypes.bool.isRequired,
	passwordLengthError: PropTypes.bool.isRequired,
	handleInputChange: PropTypes.func.isRequired,
	handleSubmit: PropTypes.func.isRequired,
};

export default Step1_CreateAccountForm;
