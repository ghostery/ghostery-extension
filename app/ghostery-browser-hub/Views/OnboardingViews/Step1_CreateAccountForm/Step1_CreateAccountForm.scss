/**
 * Step1_CreateAccountForm Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2021 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

.Step1_CreateAccountForm--addPaddingTop {
	padding-top: 20px;
}
.Step1_CreateAccountForm__inputLabel {
	font-size: 14px;
	font-weight: 500;
	line-height: 2.86;
	text-transform: uppercase;
	color: #4a4a4a;
}
.Step1_CreateAccountForm__inputBox {
	margin-bottom: 20px;
	font-size: 14;
	line-height: 24px;
	color: #4a4a4a;

	// Foundation Overrides
	border-radius: 6px;
	box-shadow: none;
	border: 1px solid #9b9b9b;

	&::placeholder {
		color: #c4ced1;
	}
}
.Step1_CreateAccountForm__inputBox.error {
	border-color: $color-create-account-form-error-red;
}
.Step1_CreateAccountForm__inputErrorContainer {
	position: relative;

	> .Step1_CreateAccountForm__inputError {
		position: absolute;
		margin-top: -15px;
		color: red;
	}
}
.Step1_CreateAccountForm__checkboxContainer {
	margin-top: 36px;
}
.Step1_CreateAccountForm__legalConsentCheckedLabel {
	font-size: 14px;
	margin-top: 13px;
	@include breakpoint(small down) {
		width: 290px;
	}

	&.error {
		color: red;
	}
}
.Step1_CreateAccountForm__ctaButtonContainer {
	display: flex;
	justify-content: center;

	.Step1_CreateAccountForm__ctaButton {
		margin: 48px auto 0 auto;
		min-height: 44px;
		width: 162px;
		padding: 7.7px 14px;
		line-height: 22px;
		background: linear-gradient(
			45deg,
			#ff7e74 50%,
			#00aef0
		);
		background-size: 200% 100%;
		background-position: 100% 50%;
		transition: 0.25s all;
		border: none;
		&:hover {
			background-position: 0% 50%;
			transition: 0.25s all;
		}
		color: #FFF;
		font-size: 14.1px;
		font-weight: 700;
		border-radius: 3.5px;
		text-align: center;
		line-height: 2.05;
		cursor: pointer;
	}
}
@media only screen and (max-width: 740px) {
	.Step1_CreateAccountForm__checkboxContainer {
		display: flex;
		align-items: center;
		padding-right: 12px;
		justify-content: center;
	}
}
