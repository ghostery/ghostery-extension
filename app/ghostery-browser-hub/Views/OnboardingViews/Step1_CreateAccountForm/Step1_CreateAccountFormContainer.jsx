/**
 * Create Account Form Container
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2020, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
	validateEmail,
	validatePassword,
	validateEmailsMatch,
	validateConfirmEmail,
	validatePasswordsMatch
} from '../../../../panel/utils/utils';
import BrowserCreateAccountForm from './Step1_CreateAccountForm';

/**
 * @class Implement the Create Account View for the Ghostery Hub
 * @extends Component
 * @memberof GhosteryBrowserHubContainers
 */
class CreateAccountFormContainer extends Component {
	constructor(props) {
		super(props);
		this.state = {
			email: '',
			emailError: false,
			confirmEmail: '',
			confirmEmailError: false,
			firstName: '',
			lastName: '',
			legalConsentChecked: false,
			legalConsentNotCheckedError: false,
			isUpdatesChecked: false,
			password: '',
			passwordInvalidError: false,
			passwordLengthError: false,
			confirmPassword: '',
			confirmPasswordError: '',
			validateInput: false,
		};

		const { actions } = this.props;
		actions.setToast({
			toastMessage: '',
			toastClass: '',
		});
	}

	/**
	 * Update input values by updating state.
	 * @param  {Object} event the 'change' event
	 */
	_handleInputChange = (event) => {
		const { name, value } = event.target;
		this.setState({ [name]: value });

		const { validateInput } = this.state;
		if (!validateInput) {
			return;
		}

		switch (name) {
			case 'email': {
				const emailIsValid = value && validateEmail(value);
				this.setState({
					emailError: !emailIsValid,
				});
				break;
			}
			case 'confirmEmail': {
				const { email } = this.state;
				const confirmIsValid = value && validateEmailsMatch(email, value);
				this.setState({
					confirmEmailError: !confirmIsValid,
				});
				break;
			}
			case 'password': {
				const passwordIsValid = value && validatePassword(value);
				const invalidChars = !passwordIsValid && value.length >= 8 && value.length <= 50;
				const invalidLength = !passwordIsValid && !invalidChars;
				this.setState({
					passwordInvalidError: invalidChars,
					passwordLengthError: invalidLength,
				});
				break;
			}
			case 'confirmPassword': {
				const { password } = this.state;
				const confirmPasswordIsValid = value && validatePasswordsMatch(password, value);
				this.setState({
					confirmPasswordError: !confirmPasswordIsValid,
				});
				break;
			}
			default: break;
		}
	}

	/**
	 * Update legal consent checkbox value by updating state
	 */
	_handleLegalConsentCheckboxChange = () => {
		this.setState(prevState => ({ legalConsentChecked: !prevState.legalConsentChecked }));
	}

	/**
	 * Update updates checkbox value by updating state
	 */
	_handleUpdatesCheckboxChange = () => {
		this.setState(prevState => ({ isUpdatesChecked: !prevState.isUpdatesChecked }));
	}

	/**
	 * Handle creating an account, but validate the data first.
	 * @param  {Object} event the 'submit' event
	 */
	_handleCreateAccountAttempt = (event) => {
		event.preventDefault();
		const {
			email,
			confirmEmail,
			firstName,
			lastName,
			legalConsentChecked,
			password,
			confirmPassword,
			isUpdatesChecked,
		} = this.state;
		const emailIsValid = email && validateEmail(email);
		const confirmIsValid = confirmEmail && validateConfirmEmail(email, confirmEmail);
		const passwordIsValid = password && validatePassword(password);
		const invalidChars = !passwordIsValid && password.length >= 8 && password.length <= 50;
		const invalidLength = !passwordIsValid && !invalidChars;
		const confirmPasswordError = password !== confirmPassword;

		this.setState({
			emailError: !emailIsValid,
			confirmEmailError: !confirmIsValid,
			legalConsentNotCheckedError: !legalConsentChecked,
			passwordInvalidError: invalidChars,
			passwordLengthError: invalidLength,
			validateInput: true,
		});

		if (!emailIsValid || !confirmIsValid || !legalConsentChecked || !passwordIsValid || confirmPasswordError) {
			return;
		}
		const { actions } = this.props;
		actions.setToast({
			toastMessage: '',
			toastClass: ''
		});
		actions.register(email, confirmEmail, firstName, lastName, password).then((success) => {
			if (success) {
				// User is automatically logged in, and redirected to the logged in view of BrowserCreateAccountForm
				actions.getUser().then(() => {
					if (isUpdatesChecked) actions.handleEmailPreferencesCheckboxChange('global', isUpdatesChecked);
				});
				// Toggle legal consent checked here
				actions.setToast({
					toastMessage: t('hub_create_account_toast_success'),
					toastClass: 'success'
				});
			} else {
				actions.setToast({
					toastMessage: t('hub_create_account_toast_error'),
					toastClass: 'alert'
				});
			}
		});
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the Create Account View of the Hub app
	 */
	render() {
		// const { user } = this.props;
		const {
			email,
			emailError,
			confirmEmail,
			confirmEmailError,
			firstName,
			lastName,
			legalConsentChecked,
			legalConsentNotCheckedError,
			isUpdatesChecked,
			password,
			passwordInvalidError,
			passwordLengthError,
			confirmPassword,
			confirmPasswordError
		} = this.state;

		return (
			<BrowserCreateAccountForm
				email={email}
				emailError={emailError}
				confirmEmail={confirmEmail}
				confirmEmailError={confirmEmailError}
				firstName={firstName}
				lastName={lastName}
				legalConsentChecked={legalConsentChecked}
				legalConsentNotCheckedError={legalConsentNotCheckedError}
				password={password}
				passwordInvalidError={passwordInvalidError}
				passwordLengthError={passwordLengthError}
				confirmPassword={confirmPassword}
				confirmPasswordError={confirmPasswordError}
				handleInputChange={this._handleInputChange}
				handleLegalConsentCheckboxChange={this._handleLegalConsentCheckboxChange}
				handleSubmit={this._handleCreateAccountAttempt}
				isUpdatesChecked={isUpdatesChecked}
				handleUpdatesCheckboxChange={this._handleUpdatesCheckboxChange}
			/>
		);
	}
}

// PropTypes ensure we pass required props of the correct type
CreateAccountFormContainer.propTypes = {
	actions: PropTypes.shape({
		setToast: PropTypes.func.isRequired,
		register: PropTypes.func.isRequired,
		getUser: PropTypes.func.isRequired,
	}).isRequired,
};

export default CreateAccountFormContainer;
