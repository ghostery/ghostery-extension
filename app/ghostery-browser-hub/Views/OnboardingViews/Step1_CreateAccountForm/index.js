/**
 * Point of entry index.js file for Ghostery Browser Hub Create Account Form
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2020 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import { buildReduxHOC } from '../../../../shared-hub/utils';
import Step1_CreateAccountFormContainer from './Step1_CreateAccountFormContainer';
import { register, getUser, handleEmailPreferencesCheckboxChange } from '../../../../Account/AccountActions';
import { setToast } from '../../../../hub/Views/AppView/AppViewActions';

const stateSlices = ['account'];
const actionCreators = {
	setToast,
	register,
	getUser,
	handleEmailPreferencesCheckboxChange
};

export default buildReduxHOC(stateSlices, actionCreators, Step1_CreateAccountFormContainer);
