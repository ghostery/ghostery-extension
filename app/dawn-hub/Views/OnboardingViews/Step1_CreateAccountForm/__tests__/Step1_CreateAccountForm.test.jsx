/**
 * Create Account View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2021 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';
import { Step1_CreateAccountForm } from '../Step1_CreateAccountForm';


jest.mock('../../../../../shared-components/ToggleCheckbox', () => {
  const ToggleCheckbox = () => <div />;
  return ToggleCheckbox;
});

const noop = () => {};
describe('app/hub/Views/Step1_CreateAccountForm component', () => {
	const initialState = {
		email: 'test@example.com',
		emailError: false,
		confirmEmail: 'test@example.com',
		confirmEmailError: false,
		firstName: 'First',
		lastName: 'Last',
		isUpdatesChecked: true,
		legalConsentChecked: true,
		password: '',
		confirmPassword: '',
		passwordInvalidError: false,
		passwordLengthError: false,
		handleInputChange: noop,
		handleUpdatesCheckboxChange: noop,
		handleLegalConsentCheckboxChange: noop,
		handleSubmit: jest.fn(),
	};

	describe('Snapshot tests with react-test-renderer', () => {
		test('create account view is rendered correctly', () => {
			const component = renderer.create(
				<MemoryRouter>
					<Step1_CreateAccountForm {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const component = shallow(<Step1_CreateAccountForm {...initialState} />);

			expect(initialState.handleSubmit.mock.calls.length).toBe(0);
			component.find('form').simulate('submit');
			expect(initialState.handleSubmit.mock.calls.length).toBe(1);
		});

		test('the sad path of the component with errors', () => {
			const initialStateFail = {
				...initialState,
				emailError: true,
				confirmEmail: 'badConfirmEmail@example.com',
				confirmEmailError: true,
				isUpdatesChecked: false,
				legalConsentChecked: false,
				password: 'password',
				confirmPassword: 'password',
				passwordInvalidError: true,
				passwordLengthError: true,
			}

			const component = shallow(<Step1_CreateAccountForm {...initialStateFail} />);
			component.find('form').simulate('submit');
			expect(component.find('.Step1_CreateAccountForm__inputErrorContainer').length).toBe(4);
			expect(component).toMatchSnapshot();

		});
	});
});
