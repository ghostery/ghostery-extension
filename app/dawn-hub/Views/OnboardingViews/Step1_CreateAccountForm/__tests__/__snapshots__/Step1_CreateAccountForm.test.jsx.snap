// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/Step1_CreateAccountForm component Shallow snapshot tests rendered with Enzyme the sad path of the component with errors 1`] = `ShallowWrapper {}`;

exports[`app/hub/Views/Step1_CreateAccountForm component Snapshot tests with react-test-renderer create account view is rendered correctly 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <div
    className="Step1_CreateAccountForm--addPaddingTop row align-center-middle"
  >
    <div
      className="columns small-10 medium-6"
    >
      <label
        className="Step1_CreateAccountForm__inputLabel"
        htmlFor="create-account-email"
      >
        <span>
          email_colon
        </span>
      </label>
      <input
        autoComplete="off"
        className="Step1_CreateAccountForm__inputBox"
        id="create-account-email"
        name="email"
        onChange={[Function]}
        pattern=".{1,}"
        placeholder="example@mail.com"
        type="text"
        value="test@example.com"
      />
    </div>
    <div
      className="columns small-10 medium-6"
    >
      <label
        className="Step1_CreateAccountForm__inputLabel"
        htmlFor="create-account-confirmEmail"
      >
        <span>
          confirm_email_colon
        </span>
      </label>
      <input
        autoComplete="off"
        className="Step1_CreateAccountForm__inputBox"
        id="create-account-confirmEmail"
        name="confirmEmail"
        onChange={[Function]}
        pattern=".{1,}"
        type="text"
        value="test@example.com"
      />
    </div>
  </div>
  <div
    className="row align-center-middle"
  >
    <div
      className="columns small-10 medium-6"
    >
      <label
        className="Step1_CreateAccountForm__inputLabel"
        htmlFor="create-account-firstName"
      >
        <span>
          hub_create_account_label_first_name
        </span>
      </label>
      <input
        autoComplete="off"
        className="Step1_CreateAccountForm__inputBox"
        id="create-account-firstName"
        name="firstName"
        onChange={[Function]}
        pattern=".{1,}"
        type="text"
        value="First"
      />
    </div>
    <div
      className="columns small-10 medium-6"
    >
      <label
        className="Step1_CreateAccountForm__inputLabel"
        htmlFor="create-account-lastName"
      >
        <span>
          hub_create_account_label_email_last_name
        </span>
      </label>
      <input
        autoComplete="off"
        className="Step1_CreateAccountForm__inputBox"
        id="create-account-lastName"
        name="lastName"
        onChange={[Function]}
        pattern=".{1,}"
        type="text"
        value="Last"
      />
    </div>
  </div>
  <div
    className="row align-center-middle"
  >
    <div
      className="columns small-10 medium-6"
    >
      <label
        className="Step1_CreateAccountForm__inputLabel"
        htmlFor="create-account-password"
      >
        <span>
          password_colon
        </span>
      </label>
      <input
        autoComplete="off"
        className="Step1_CreateAccountForm__inputBox"
        id="create-account-password"
        name="password"
        onChange={[Function]}
        pattern=".{1,}"
        type="password"
        value=""
      />
    </div>
    <div
      className="columns small-10 medium-6"
    >
      <label
        className="Step1_CreateAccountForm__inputLabel"
        htmlFor="create-account-password"
      >
        <span>
          confirm_password_colon
        </span>
      </label>
      <input
        autoComplete="off"
        className="Step1_CreateAccountForm__inputBox"
        id="create-account-password"
        name="confirmPassword"
        onChange={[Function]}
        pattern=".{1,}"
        type="password"
        value=""
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="columns small-12 medium-10"
    >
      <div
        className="Step1_CreateAccountForm__checkboxContainer BrowserCreateAccountForm--marginBottom flex-container"
      >
        <div />
        <span
          className="Step1_CreateAccountForm__legalConsentCheckedLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "create_account_form_legal_consent_checkbox_label",
            }
          }
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="Step1_CreateAccountForm__ctaButtonContainer"
  >
    <button
      className="Step1_CreateAccountForm__ctaButton"
      type="submit"
    >
      create_account
    </button>
  </div>
</form>
`;
